bodhi/server/services/updates.py

```diff
@@ -189,6 +194,36 @@ def set_request(request):
     return dict(update=update)
 
 
+@update_rerun_test.post(schema=bodhi.server.schemas.RerunTestSchema,
+                     validators=(
+                         validate_update_id,
+                         validate_acls,
+                     ),
+                     permission='edit', renderer='json',
+                     error_handler=bodhi.server.services.errors.json_handler)
+def rerun_test(request):
+    """
+    Send fedmsg request that a specific test for a specific update be re-run.
+
+    Args:
+        request (pyramid.request): The current request.
+    Returns:
+        dict: A dictionary with at least the following key mappings:
+            update: The update for which the rerun was requested.
+            test: The test for which the rerun was requested.
+            scenario: The scenario for which the rerun was requested.
+    """
+    data = request.validated
+    update = data['update']
+    test = data['test']
+    scenario = data['scenario']
+    log.info("Requesting rerun of test %s, scenario %s for update %s",
+             test, scenario, update)
```

Comment:
Optional: It might be useful to include the username in this log message. Also optional: consider debug level since the fedmsg itself will also be trackable.